xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecAndroidMediaCodec.cpp

```diff
@@ -399,16 +399,30 @@ bool CDVDAudioCodecAndroidMediaCodec::AddData(const DemuxPacket &packet)
         return false;
 
       CJNIMediaCodecCryptoInfo *cryptoInfo(0);
+      CJNIMediaCodecCryptoInfoPattern* cryptoInfoPattern(nullptr);
```

Comment:
since his is stored in the class, why not use a unique_ptr?

Reply:
Please forgive me here, still quite new with c++...
I've followed in the example of the existing code and libandroidjni but I can't figure out how to make the necessary changes in the libandroidjni code for unique_ptr.

Reply:
@enen92 I've looked into this more - I don't think this is possible without big changes in libandroidjni (and even then I'm not sure, it's all tied to JNI/NDK) which is outside the scope of this PR. None of the other objects in this file are treated as such either.

Reply:
nvm, I was doing it wrong!